### PR TITLE
Updated ckwriter offset calculations and added spkwriter calculations

### PR DESCRIPTION
**Pull request by amystamile-usgs**
## Description
In spiceinit, Kernels are matched to the cube based on the original labels start and end times.

This comparison checks if the kernel start time is less than the label start time and the kernel end time is greater than the label end time. Therefore checking that the label's start and end time lie between the kernel's start and end time.

`kernelStartTime <= labelStartTime  < labelEndTime <= kernelEndTime`

Therefore when checking the offsets, there is only a need to apply an offset if the label's start time is less than the camera start time and if the label's end time is greater than the kernel's end time. 

Also found out that these offsets are not the same for both start time and end time. Therefore this PR adds the end time offset to the comments as well.

```
Object = SpacecraftPosition
  RunTime = 2021-12-27T11:34:33

  Group = Dependencies
    LeapsecondKernel = $Base/kernels/lsk/naif0008.tls
  End_Group

  Group = Selection
    Time        = ("2002 FEB 20 22:59:01.701369 TDB",
                   "2002 FEB 20 22:59:09.296828 TDB")
    Instrument  = THEMIS_IR
    StartOffset = 0.26313
    EndOffset   = 171.871
    File        = /Users/astamile/testData/kernelTiming/thmIR.bsp
    Type        = Smithed
  End_Group
End_Object
End
```

These offsets are in absolute values therefore if there is a StartOffset, then this must be added to the label's original start time and if there is a EndOffset, then the offset must be subtracted from the label's original end time.

## Motivation and Context

Kernel Timing sprint

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/apps/ckwriter/CkSpiceSegment.cpp [hunk 1]
```diff
@@ -265,9 +267,21 @@ void CkSpiceSegment::import(Cube &cube, const QString &tblname) {
     _utcStartTime = toUTC(startTime());
     _utcEndTime   = toUTC(endTime());
 
-    _timeOffset =  fabs(etLabStart.Et() - startTime());
+    _startOffset =  etLabStart.Et() - startTime();
+    _endOffset =  etLabEnd.Et() - endTime();
     // account for padding
-    if (_timeOffset <= 0.003) { _timeOffset = 0; }
+    if (_startOffset >= 0.0031) {
```
Comment (AustinSanders) [inline isis/src/base/apps/ckwriter/CkSpiceSegment.cpp]: If the padding is .003, should this just be > .003 instead of >=.0031?

Comment (amystamile-usgs) [reply isis/src/base/apps/ckwriter/CkSpiceSegment.cpp]: @AustinSanders  I was running into a situation in which the offset was 0.00305282 I guess due to the precision of doubles? Would a better solution be to set the precision?

Comment (AustinSanders) [reply isis/src/base/apps/ckwriter/CkSpiceSegment.cpp]: Hmm.  I'm not really sure.  With the current solution, what happens if the offset falls between .003 and .0031?

Comment (amystamile-usgs) [reply isis/src/base/apps/ckwriter/CkSpiceSegment.cpp]: Fixed this check by first rounding offsets to three decimal places before doing the padding check. It is necessary that these offsets are within three decimals anyways for when we add/subtract the label time with these offsets.

**Review by AustinSanders — APPROVED** [review]

**Pull request merged**